### PR TITLE
QuantileExactExclusive & Inclusive function added.

### PR DESCRIPTION
**Pull request by dimarub2000**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- New Feature

Short description (up to few sentences):
Suggested in #5885 and implemented in a better way.

### CODE REVIEW

**Comment by dimarub2000** [comment]
I am thinking about adding an inclusive version, and probably improve semantics with QuantileExactBase class.

File: dbms/src/AggregateFunctions/QuantileExact.h [hunk 1]
```diff
@@ -64,7 +64,7 @@ struct QuantileExact
     }
 
     /// Get the value of the `level` quantile. The level must be between 0 and 1.
-    Value get(Float64 level)
+    virtual Value get(Float64 level)
```
Comment (alexey-milovidov) [inline dbms/src/AggregateFunctions/QuantileExact.h]: Need performance test with many (millions) of aggregate function states.

File: dbms/src/AggregateFunctions/QuantileExact.h [hunk 2]
```diff
@@ -108,15 +108,77 @@ struct QuantileExact
...
+                auto n = static_cast<size_t>(h);
+
+                if (n >= array.size())
+                    result[indices[i]] = array[array.size() - 1];
+                else if (n < 1)
+                    result[indices[i]] = array[0];
+                else
+                {
+                    std::nth_element(array.begin() + prev_n, array.begin() + n - 1, array.end());
+                    std::nth_element(array.begin() + n, array.begin() + n, array.end());
```
Comment (alexey-milovidov) [inline dbms/src/AggregateFunctions/QuantileExact.h]: Just minimum?

File: dbms/src/AggregateFunctions/QuantileExact.h [hunk 3]
```diff
@@ -108,15 +108,77 @@ struct QuantileExact
...
+    virtual void getManyFloat(const Float64 *, const size_t *, size_t, Float64 *)
     {
         throw Exception("Method getManyFloat is not implemented for QuantileExact", ErrorCodes::NOT_IMPLEMENTED);
     }
+
+    virtual ~QuantileExact() = default;
+};
+
+template <typename Value>
+struct QuantileExactExclusive : public QuantileExact<Value>
```
Comment (alexey-milovidov) [inline dbms/src/AggregateFunctions/QuantileExact.h]: Missing comment.

File: dbms/src/AggregateFunctions/QuantileExact.h [hunk 4]
```diff
@@ -81,7 +81,7 @@ struct QuantileExact
 
     /// Get the `size` values of `levels` quantiles. Write `size` results starting with `result` address.
     /// indices - an array of index levels such that the corresponding elements will go in ascending order.
-    void getMany(const Float64 * levels, const size_t * indices, size_t size, Value * result)
+    virtual void getMany(const Float64 * levels, const size_t * indices, size_t size, Value * result)
```
Comment (alexey-milovidov) [inline dbms/src/AggregateFunctions/QuantileExact.h]: I don't want to pay for extra 8 bytes.

File: dbms/src/AggregateFunctions/QuantileExact.h (hunk 1 above)
Comment (dimarub2000) [reply dbms/src/AggregateFunctions/QuantileExact.h]: > Need performance test with many (millions) of aggregate function states.

Some of `virtual`'s were redundant in this commit. It's needed only for `getFloat` and `getManyFloat`, so the perfomance of old function won't be affected.

File: dbms/src/AggregateFunctions/QuantileExact.h (hunk 4 above)
Comment (alexey-milovidov) [reply dbms/src/AggregateFunctions/QuantileExact.h]: The existence of at least one virtual method will add extra 8 bytes to the sizeof struct.

Comment (dimarub2000) [reply dbms/src/AggregateFunctions/QuantileExact.h]: Actually functions `getFloat` and `getManyFloat` were unused and throwing exceptions. One solution is to delete them as I did in a latest commit.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**